ui/src/state/groups/groups.ts

```diff
@@ -47,1094 +40,1227 @@ function groupAction(flag: string, diff: GroupDiff) {
+              }
 
-  Object.entries(gangs)
-    .filter(([flag, g]) => g.invite !== null && !(flag in groups))
-    .filter(
-      ([, gang]) =>
-        !gang.claim ||
-        gang.claim.progress === 'error' ||
-        gang.claim.progress === 'knocking'
-    )
-    .forEach(([flag, gang]) => {
-      pendingGangs[flag] = gang;
+              return false;
+            });
+          resolve();
+        },
+      });
     });
+  };
 
-  return pendingGangs;
+  return useGroupMutation(mutationFn);
 }
 
-const selInit = (s: GroupState) => s.initialized;
-export function useGroupsInitialized() {
-  return useGroupState(selInit);
+export function useGroupDelMembersMutation() {
+  const mutationFn = async (variables: { flag: string; ships: string[] }) => {
+    const diff = {
+      fleet: {
+        ships: variables.ships,
+        diff: {
+          del: null,
+        },
+      },
+    };
+    await api.poke(groupAction(variables.flag, diff));
+  };
+
+  return useGroupMutation(mutationFn);
 }
 
-export function useSects(flag: string) {
-  const group = useGroup(flag);
-  return group ? Object.keys(group.cabals) : [];
+export function useGroupAddRoleMutation() {
+  // Not used yet.
+  const mutationFn = async (variables: {
+    flag: string;
+    sect: string;
+    meta: GroupMeta;
+  }) => {
+    const diff = {
+      cabal: {
+        sect: variables.sect,
+        diff: {
+          add: { ...variables.meta, image: '', cover: '' },
+        },
+      },
+    };
+    await api.poke(groupAction(variables.flag, diff));
+  };
+
+  return useGroupMutation(mutationFn);
 }
 
-const { shouldLoad, newAttempt, finished } = getPreviewTracker();
+export function useGroupDelRoleMutation() {
+  // Not used yet.
+  const mutationFn = async (variables: { flag: string; sect: string }) => {
+    const diff = {
+      cabal: {
+        sect: variables.sect,
+        diff: { del: null },
+      },
+    };
+    await api.poke(groupAction(variables.flag, diff));
+  };
 
-export function useChannelPreview(nest: string, disableLoading = false) {
-  const preview = useGroupState(
-    useCallback((s) => s.channelPreviews[nest], [nest])
-  );
+  return useGroupMutation(mutationFn);
+}
 
-  const getChannelPreview = useCallback(async () => {
-    asyncCallWithTimeout(
-      useGroupState.getState().channelPreview(nest),
-      10 * 1000
-    ).finally(() => finished(nest));
-  }, [nest]);
+export function useGroupIndex(ship: string) {
+  const { data, ...rest } = useReactQuerySubscribeOnce({
+    queryKey: ['group-index', ship],
+    app: 'groups',
+    path: `/gangs/index/${ship}`,
+    options: {
+      enabled: ship !== '',
+    },
+  });
 
-  useEffect(() => {
-    if (disableLoading || (preview && !shouldLoad(nest))) return;
+  if (rest.isLoading) {
+    return null;
+  }
 
-    newAttempt(nest);
-    getChannelPreview();
-  }, [getChannelPreview, preview, nest, disableLoading]);
+  if (rest.isError) {
+    throw new Error('Failed to fetch group index');
```

Comment:
when used like a normal hook this means if it fails we'll hit the error boundary and they'll have to refresh